integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -401,18 +400,22 @@ public int release(String path, FuseFileInfo fi) {
 
   private int releaseInternal(String path, FuseFileInfo fi) {
     long fd = fi.fh.get();
-    try (LockResource r1 = new LockResource(mFileLocks.get(fd).writeLock())) {
+    try {
       FileInStream is = mOpenFileEntries.remove(fd);
       FileOutStream os = mCreateFileEntries.remove(fd);
       if (is == null && os == null) {
         LOG.error("Cannot find fd {} for {}", fd, path);
         return -ErrorCodes.EBADFD();
       }
       if (is != null) {
-        is.close();
+        synchronized (is) {
+          is.close();
```

Comment:
Do we have to check if `is` is in the index? (same question for `os` below)

Reply:
if `mOpenFiles.remove(oe)` return true, it means other threads cannot remove successfully and return false, so that other threads will return with `-ErrorCodes.EBADFD()`.

And for the lucky thread which remove the oe successfully and return true, it can step into the synchronized block if no other threads was reading from the inputStream, so `oe.close` can be executed only one time.

Reply:
sounds good